### PR TITLE
fix: window icons on linux

### PR DESCRIPTION
**Pull request by st3iny**
### ☑️ Resolves

* Issue #53

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
B | A
| grafik | grafik |

### 🚧 Tasks

- [ ] ...

### CODE REVIEW

File: src/shared/icons.utils.js [hunk 1]
```diff
@@ -58,6 +63,26 @@ function getTrayIcon(platform, theme) {
...
+		// application has to configure it manually. Othwerwise, the default icon of the window
+		// manager will be shown. The icon should preferably be a png.
+		return {
+			icon: icons.window.linux,
+		}
+	}
+
+	// The icon is already handled on other platforms.
+	return {}
+}
```
Comment (ShGKme) [inline src/shared/icons.utils.js:83]: As icons are guaranteed to be only one property `icon`, I'd return the icon path here, not an object with one property, similar to `getTrayIcon`.

It is safe to return `undefined` to use the executable's icon on Mac and Windows.

File: src/shared/icons.utils.js [hunk 2]
```diff
@@ -20,8 +20,13 @@
  */
 
 const { nativeTheme } = require('electron')
+const { isLinux } = require('./os.utils.js')
 
 const icons = {
+	window: {
+		linux: require('../../img/icons/icon.png'),
+	},
```
Comment (ShGKme) [inline src/shared/icons.utils.js:28]: I'd add here a comment here: `Executable's icon is used by default`

**Review by ShGKme — CHANGES_REQUESTED** [review]
Hi Richard and thank you for the fix! I totally forgot that after fixing the problem with the tray icon, it should also work now with the browser window (at least in some desktop environment) :D 

But I'd make `getBrowserWindowIconOptions` an implicit `getBrowserWindowIcon` function to return the icon, because it is only used for setting icon.

**Comment by Antreesy** [comment]
Still don't have an icon in package :eyes: 
Using Ubuntu 22.04.3 LTS

image

**Comment by st3iny** [comment]
> Still don't have an icon in package eyes Using Ubuntu 22.04.3 LTS

I can confirm this. For some reason it works when I run the packaged binary in `out/Nextcloud Talk-linux-x64/Nextcloud Talk` but when I move/install it somewhere else it doesn't work anymore.

File: src/shared/icons.utils.js (hunk 1 above)
Comment (st3iny) [reply src/shared/icons.utils.js]: Done

File: src/shared/icons.utils.js (hunk 2 above)
Comment (st3iny) [reply src/shared/icons.utils.js:30]: Done

**Comment by st3iny** [comment]
I figured it out. The path to the icon should be absolute so that it works regardless of where the app is installed and run from.

@ShGKme @Antreesy Could you please give this another test. Thanks :)

**Review by nickvergessen — APPROVED** [review]
Works perfectly now :)

**Review by ShGKme — APPROVED** [review]
Cool!

**Pull request merged**